File: scratch/dump_schema.js
const sqlite3 = require('sqlite3').verbose();
const db = new sqlite3.Database('c:/Users/JULIAN/Desktop/GravyLocal2.0/pb_data/data.db');

db.all("SELECT name, listRule, viewRule, createRule, updateRule, deleteRule, fields FROM _collections", (err, rows) => {
  if (err) {
    console.error(err);
  } else {
    for (const row of rows) {
      console.log(`\n=========================================`);
      console.log(`Collection: ${row.name}`);
      console.log(`Rules: list=${row.listRule}, view=${row.viewRule}, create=${row.createRule}, update=${row.updateRule}, delete=${row.deleteRule}`);
      try {
        const fields = JSON.parse(row.fields);
        console.log(`Fields:`);
        fields.forEach(f => {
          console.log(`  - ${f.name} (${f.type})${f.required ? ' *' : ''}`);
        });
      } catch (e) {
        console.log(`Fields (raw): ${row.fields}`);
      }
    }
  }
  db.close();
});

File: scratch/dump_schema.py
import sqlite3
import json

db_path = r'c:\Users\JULIAN\Desktop\GravyLocal2.0\pb_data\data.db'
conn = sqlite3.connect(db_path)
cursor = conn.cursor()

cursor.execute("SELECT name, listRule, viewRule, createRule, updateRule, deleteRule, fields FROM _collections")
rows = cursor.fetchall()

for row in rows:
    name, list_rule, view_rule, create_rule, update_rule, delete_rule, fields_json = row
    print("=" * 50)
    print(f"Collection: {name}")
    print(f"Rules: list={list_rule}, view={view_rule}, create={create_rule}, update={update_rule}, delete={delete_rule}")
    try:
        fields = json.loads(fields_json)
        print("Fields:")
        for f in fields:
            name_f = f.get('name')
            type_f = f.get('type')
            req = '*' if f.get('required') else ''
            print(f"  - {name_f} ({type_f}){req}")
    except Exception as e:
        print(f"Fields (error parsing): {fields_json}")

conn.close()
